src/sdk/makeDimensions.js

```diff
@@ -126,7 +126,9 @@ export default chart => {
   const getDimensionColor = id => {
     const colors = chart.getAttribute("colors")
     const { context } = chart.getMetadata()
-    const key = colors.length ? chart.getAttribute("id") : context
+    const groupBy = chart.getAttribute("groupBy")
+    const key =
+      groupBy === "dimension" ? (colors.length ? chart.getAttribute("id") : context) : groupBy
```

Comment:
Could move `(colors.length ? chart.getAttribute("id") : context)` to a variable to avoid nested ternaries + add some context, for readability?